src/type.js

```diff
@@ -4,44 +4,121 @@ import { core } from 'metal';
 
 const validators = {
 	any: () => true,
-	array: primitiveTypeValidatorFactory('array'),
-	bool: primitiveTypeValidatorFactory('boolean'),
-	func: primitiveTypeValidatorFactory('function'),
-	number: primitiveTypeValidatorFactory('number'),
-	object: primitiveTypeValidatorFactory('object'),
-	string: primitiveTypeValidatorFactory('string'),
-
-	arrayOf: arrayOfValidatorFactory,
-	instanceOf: instanceOfValidatorFactory,
-	objectOf: objectOfValidatorFactory,
-	oneOfType: oneOfTypeValidatorFactory,
-	shapeOf: shapeOfValidatorFactory
-};
-
-/**
- * Creates a validator that checks the values of an array against a type.
- * @param {!function} validator Type validator to check each index against.
- * @return {function} Validator.
- */
-function arrayOfValidatorFactory(validator) {
-	return (value, name, context) => {
-		if (!Array.isArray(value)) {
-			return composeError('Expected an array.', name, context);
-		} else {
-			const testArray = value.every(
-				item => {
-					return !(validator(item, name) instanceof Error);
+	array: validateType('array'),
+	bool: validateType('boolean'),
+	func: validateType('function'),
+	number: validateType('number'),
+	object: validateType('object'),
+	string: validateType('string'),
+
+	/**
+	 * Creates a validator that checks the values of an array against a type.
+	 * @param {!function} validator Type validator to check each index against.
```

Comment:
Perfect, thanks! I think this is easy to read now :D

The only thing I've noticed now is that the doc params are not entirely following what we've been using (which is fine, since that's not documented anywhere yet). We're using google's standard for this, which you can check out [here](https://developers.google.com/closure/compiler/docs/js-for-compiler#types).

For example, functions are usually written like `function()` and arrays and objects use uppercase like `Array` and `Object`. You're already using `!` to indicate that a value shouldn't be null, it's just missing in the `@return` expressions too. In this function's case for example it should be `@return {!function()} Validator`.

I could fix these minor doc issues for you, but I think it helps to remember these things when we do it ourselves, specially when first getting used to a repo's standards.